langext/src/countdown.c

```diff
@@ -49,6 +49,10 @@ int rage_countdown_add(rage_Countdown * c, int delta) {
     return val;
 }
 
-void rage_countdown_unblock_wait(rage_Countdown * c) {
-    sem_post(&c->sig);
+int rage_countdown_unblock_wait(rage_Countdown * c) {
+    int val = atomic_exchange(&c->counter, 0);
```

Comment:
This is a little mysterious for me - googling turns up a lot of C++! Remind me to ask you about it on Monday.

Reply:
It puts 0 in the counter and gives you a copy of the previous value atomically.